test/__snapshots__/main.test.ts.snap

```diff
@@ -1920,174 +1922,190 @@ export interface RouteConfig {
 export default function getRoutes(_metarpheusRouteConfig: RouteConfig) {
   return {
     /** Create a new search query for available vehicles */
-    availableVehicleController_createQuery: function ({ query }: { query: m.AvailableVehicleSearchQuery }): TaskEither<unknown, m.UUID> {
-      return tryCatch(() => axios({
-        method: 'post',
-        url: \`\${_metarpheusRouteConfig.apiEndpoint}/availableVehicles/createQuery\`,
-        params: {
-
-        },
-        data: {
+    availableVehicleController_createQuery: function ({ query }: { query: m.AvailableVehicleSearchQuery }): TE.TaskEither<unknown, m.UUID> {
+      return pipe(
+        TE.tryCatch(() => axios({
+          method: 'post',
+          url: \`\${_metarpheusRouteConfig.apiEndpoint}/availableVehicles/createQuery\`,
+          params: {
+
+          },
+          data: {
           query: m.AvailableVehicleSearchQuery.encode(query)
-        },
-        headers: {
+          },
+          headers: {
           'Content-Type': 'application/json'
-        },
-        timeout: _metarpheusRouteConfig.timeout
-      }), identity).chain(res =>
-              fromEither(m.UUID.decode(res.data))
-            )
+          },
+          timeout: _metarpheusRouteConfig.timeout
+        }), (v): unknown => v),
+        TE.map(res => res.data),
+        TE.chainEitherK(flow(m.UUID.decode, E.mapLeft((e): unknown => e)))
```

Comment:
why do we need `E.mapLeft((e): unknown => e)` here? is it to upcast the type of left?

Reply:
yes, it's a `.widen` more or less. I'm not really sure why it wasn't needed before though, I tried a few versions of this `pipe`, but all required some sort of upcast to compile